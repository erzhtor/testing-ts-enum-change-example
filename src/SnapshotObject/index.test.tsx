import { render } from "@testing-library/react";
import { SnapshotObject } from ".";

enum ExampleEnum {
    One,
    Two,
    Three,
}

enum ExampleEnum2 {
    Four = "One",
    Five = "Five",
    Six = "Six",
}

describe("test enums", () => {
    test("SnapshotObject", () => {
        const { asFragment } = render(<SnapshotObject object={ExampleEnum} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("ExampleEnum2", () => {
        const { asFragment } = render(<SnapshotObject object={ExampleEnum2} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
