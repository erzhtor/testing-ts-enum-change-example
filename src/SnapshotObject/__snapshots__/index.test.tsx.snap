// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test enums ExampleEnum2 1`] = `
<DocumentFragment>
  ["Four","Five","Six"]
</DocumentFragment>
`;

exports[`test enums SnapshotObject 1`] = `
<DocumentFragment>
  ["0","1","2","One","Two","Three"]
</DocumentFragment>
`;
